src/main/scala/zio/codec/CharCodecModule.scala

```diff
@@ -5,14 +5,16 @@ import zio.internal.Stack
 
 trait CharCodecModule extends CodecModule {
   type Input = Char
-
   // todo: consumeN
-  def token(t: String): Codec[Chunk[Char]] =
+  def token(t: String): Codec[Chunk[Input]] =
     consume.repN(t.length).filter(Set(Chunk.fromIterable(t)))
 
   def tokenAs[A](t: String, v: A): Codec[A] =
     token(t).as(v, Chunk.fromIterable(t))
 
+  def betweenChar(begin: Input, end: Input): Codec[Input] =
+    consume.filter((begin.toInt to end.toInt).map(_.toChar).toSet)
```

Comment:
I think you dont need to do the conversion to int here. you can call `begin to end`

Reply:
It's because `begin` and `end` are `Char`, and Range works with `Int`

Reply:
since scala compilation warning is activate `warn-numeric-widen` it is mandatory to do it

Reply:
@regis-leray huh weird. i just tested on my zio-codec fork and it and it worked fine. @sergei-shabanau char is a range as well. if you do 'a' to 'z' to get `a,b,c,d,...,z `

Reply:
https://www.cs.cmu.edu/~pattis/15-1XX/common/handouts/ascii.html

thats the equivalent table so if oyu do ` ' ' to '~'` you get all the ascii chars

Reply:
@sken77 @regis-leray 
It's a rabbit hole 🐰

We defined 
```scala
def repRange(range: Range): Codec[Chunk[A]]
```

However
![image](https://user-images.githubusercontent.com/12940501/79627295-11b33c00-8105-11ea-93bf-da2b4b5dbbad.png)

And if we dig deeper, we see a comment in `NumericRange.scala` from Paul Phillips
```
// TODO: Now the specialization exists there is no clear reason to have
// separate classes for Range/NumericRange.  Investigate and consolidate.
```

Never happened in Scala.
So maybe in Dotty 🤷‍♂️
Or ZIO 😜

Reply:
oh yeah in that case it wont work. i was talking about the example above since it just converts the range to a set

Reply:
thank @sken77 much compact `(begin to end).toSet` and no implicit conversion when using operator range `to`, everything is type check.